core/tests/test_views.py

```diff
@@ -177,3 +176,49 @@ def test_add_deeply_nested_comment_valid(client: Client, another_user: User, pos
     assert post.children_count == 10
     assert parent_comment.children_count == 0
     assert parent_comment.parent.children_count == 1
+
+
+@pytest.mark.django_db()
+def test_restricted_community_access(client: Client, restricted_community: Community, user: User) -> None:
+    client.force_login(user)
+    response = client.get(reverse("community-detail", kwargs={"slug": restricted_community.slug}))
+    assert response.status_code == 200
+
+
+def test_create_community_view(client: Client, user: User) -> None:
+    client.force_login(user)
+    url = reverse("community-create")
+    data = {
+        "name": "Test Community",
+        "privacy": "10_PUBLIC",
+        "is_18_plus": True,
+    }
+
+    response = client.post(url, data)
+    assert response.status_code == 302
+
+    community = Community.objects.get(name="Test Community")
+    assert community.name == "Test Community"
+    assert community.author == user
+    assert community.privacy == "10_PUBLIC"
+    assert community.is_18_plus is True
+
+
+@pytest.mark.django_db()
+def test_community_detail_view(client: Client, user: User, community: Community) -> None:
+    client.force_login(user)
+    url = reverse("community-detail", kwargs={"slug": community.slug})
+    response = client.get(url)
+
+    assert response.status_code == 200
+    assert "community" in response.context
+    assert response.context["community"] == community
+
+
+@pytest.mark.django_db()
+def test_update_community_view_without_permission(client: Client, community: Community, user: User) -> None:
+    client.force_login(user)
+    response = client.post(reverse("community-update", kwargs={"slug": community.slug}), {"name": "Updated Community"})
+    assert response.status_code == 403
```

Comment:
hmm 403 would be good if this is API view. 
Look how admin page behaves: http://127.0.0.1:8000/admin/login/?next=/admin/ - try to log in with fake data. 
It returns 200 OK but there is an error message. 
With 403 django shows default permision denied page, which we don't want, because it is not user friendly and user has no idea what hit him and why.
https://docs.djangoproject.com/en/5.0/ref/views/#the-403-http-forbidden-view

You can use this function: https://docs.djangoproject.com/en/5.0/topics/auth/default/#django.contrib.auth.mixins.AccessMixin.handle_no_permission to redirect back with error message. 

Or am I not seeing something here?

Reply:
@memorisanka has this been implemented?

Reply:
I'm on it. I've made some changes but they are not pushed. I had some problem with tests but had no time to solve it. I'll do it until the end of this week.

Reply:
Thanks, I got confused and noticed it was resolved.